internal/nexodus/utils_darwin.go

```diff
@@ -3,15 +3,49 @@
 package nexodus
 
 import (
+	"bytes"
 	"fmt"
-	"github.com/vishvananda/netlink"
-	"go.uber.org/zap"
 	"net"
+	"os"
+	"os/exec"
 	"strings"
+
+	"github.com/vishvananda/netlink"
+	"go.uber.org/zap"
 )
 
-// RouteExistsOS currently only used for darwin build purposes
-func RouteExistsOS(s string) (bool, error) {
+// RouteExistsOS checks to see if a route exists for the specified prefix
+func RouteExistsOS(prefix string) (bool, error) {
+	if err := ValidateIp(prefix); err != nil {
+		return false, err
+	}
+
+	r, w, err := os.Pipe()
+	if err != nil {
+		return false, err
+	}
+	defer r.Close()
+	ns := exec.Command("netstat", "-r", "-n")
+	ns.Stdout = w
+	if err = ns.Start(); err != nil {
+		return false, err
+	}
+	defer func() {
+		_ = ns.Wait()
+	}()
+	w.Close()
+
+	// #nosec -- G204: Subprocess launched with a potential tainted input or cmd arguments (gosec)
```

Comment:
Definitely fair, i thought because im doing the validation above then it's alright to keep this otherwise i would have looked for a different approach but not too opinionated on this.